### DISCUSSION TITLE
How to use plain UART?

### DISCUSSION THREAD

**Discussion by alexander-zylla** [mwaylabs/fruitymesh#204] [Q&A]
I am trying to use FruityMesh to make an existing product use Bluetooth Mesh. I want to be able to implement a UART-based Modbus protocol on the FruityMesh chip. Sadly I only found Uart examples that use the Terminal Mode. Is it possible to use the FruityMesh UART in plain mode to send actual raw byte data?

**Comment by mariusheil** [comment]
Hello,

yes, this is certainly possible. Actually, we have just been working on exactly this topic and have integrated Modbus over RS-232 and over RS-485 into our commercial offering BlueRange. So basically, we now offer a full solution of reading / writing data to Modbus devices where a FruityMesh controller acts as a Modbus master. If you are interested, you can write to "info@bluerange.io". Our platform can then be used to read from and to write to these devices and extract sensor information in a generic way.

If you want to implement a custom UART (which you will need to deal with e.g. the timings of Modbus), you have to disable the FruityMesh Terminal by using "#define ACTIVATE_UART 0" or by not defining it at all. This is a bit counterintuitive as it should be more sth. like "ACTIVATE_FRUTIYMESH_UART_TERMINAL".

You can then e.g. use the nordic drivers to implement the UART in a way that you like. E.g. you can look at how the terminal implements the functionality or look at the Nordic examples.

Marius
